WordPress/gutenberg
# Placeholder: Convert component to TypeScript

## What?
Refactor the `Placeholder` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactored the component TypeScript
- Refactored storybook to typescript and controls

_Note_
Tests are not converted to TypeScript since it uses `enzyme` and need to be refactored. Will do that in a follow-up PR.

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Test the Placeholder in the storybook.
3. Test the various use of Placeholders within the block/site editors in Gutenberg still works
